algorithms/src/hash_to_curve/hash_to_curve.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::crypto_hash::Blake2Xs;
+use snarkvm_curves::AffineCurve;
+
+/// Attempts to run hash-to-curve and returns the generator, message, and counter on sucess.
+#[inline]
+pub fn try_hash_to_curve<G: AffineCurve>(input: &str) -> Option<(G, String, usize)> {
+    // Attempt to increment counter `k` at most `8 * G::SERIALIZED_SIZE` times.
+    for k in 0..(8 * G::SERIALIZED_SIZE) {
+        // Construct a new message.
+        let message = format!("{} in {}", input, k);
+
+        // Output the generator if a valid generator was found.
+        if let Some(g) = hash_to_curve::<G>(&message) {
+            return Some((g, message, k));
+        }
+    }
+    None
+}
```

Comment:
Do we expect in practice that we don't find a curve point within 8 * 32 iterations? Also, why is `8 * SERIALIZED_SIZE` a good number?

Reply:
I believe the standard assumption is to find a solution within `k` bits iterations where `k := log2(modulus)`

Reply:
"To control bias, hash_to_field instead uses random integers whose
   length is at least ceil(log2(p)) + k bits, where k is the target
   security level for the suite in bits.  Reducing such integers mod p
   gives bias at most 2^-k for any p; this bias is appropriate when
   targeting k-bit security.
"
From Section 5.1 in https://datatracker.ietf.org/doc/draft-irtf-cfrg-hash-to-curve/

Reply:
I would use `BaseField::MODULUS_BITS` in that case

Reply:
From reading the PR, and based on the rejection probability in each iteration, I think we can just set the upper bound to some number like 128 or 256.

Reply:
Discussed offline, will use a for loop to `128` and `panic!` on failure